### PR TITLE
Clarify voting-only master node docs

### PR DESCRIPTION
**Pull request by ywelsch**
Clarifies the roles of a dedicated voting-only master-eligible node.

### CODE REVIEW

File: docs/reference/modules/node.asciidoc [hunk 1]
```diff
@@ -230,16 +257,18 @@ To create a dedicated ingest node in the {default-dist}, set:
...
 -------------------
 node.master: false <1>
-node.data: false <2>
-node.ingest: true <3>
-node.ml: false <4>
+node.voting_only: false <2>
+node.data: false <3>
+node.ingest: true <4>
+node.ml: false <5>
 cluster.remote.connect: false <5>
```
Comment (jrodewig) [inline docs/reference/modules/node.asciidoc]: This callout is causing the docbldesx CI to fail. Here's a fix:
```suggestion
cluster.remote.connect: false <6>
```

**Review by jrodewig — APPROVED** [review]
Left a fix for the CI. Otherwise LGTM.

File: docs/reference/modules/node.asciidoc [hunk 2]
```diff
@@ -176,6 +178,29 @@ reasonably fast persistent storage and a reliable and low-latency network
 connection to the rest of the cluster, since they are on the critical path for
 <<cluster-state-publishing,publishing cluster state updates>>.
 
+Similar to dedicated master-eligible nodes, it is a good idea in bigger
```
Comment (DaveCTurner) [inline docs/reference/modules/node.asciidoc]: Alternative wording suggestion:

```
Voting-only master-eligible nodes may also fill other roles in your cluster.
For instance, a node may be both a data node and a voting-only master-eligible
node. A _dedicated_ voting-only master-eligible nodes is a voting-only
master-eligible node that fills no other roles in the cluster. To create a
dedicated voting-only master-eligible node in the {default-dist}, set:
```

Comment (ywelsch) [reply docs/reference/modules/node.asciidoc]: thanks, applied in 2fc4803

**Review by DaveCTurner — APPROVED** [review]
LGTM2

File: docs/reference/modules/node.asciidoc [hunk 3]
```diff
@@ -176,6 +178,30 @@ reasonably fast persistent storage and a reliable and low-latency network
 connection to the rest of the cluster, since they are on the critical path for
 <<cluster-state-publishing,publishing cluster state updates>>.
 
+Voting-only master-eligible nodes may also fill other roles in your cluster.
+For instance, a node may be both a data node and a voting-only master-eligible
+node. A _dedicated_ voting-only master-eligible nodes is a voting-only
+master-eligible node that fills no other roles in the cluster. To create a
+dedicated voting-only master-eligible node in the {default-dist}, set:
```
Comment (zuketo) [inline docs/reference/modules/node.asciidoc]: This language is great

**Review by zuketo — APPROVED** [review]
LGTM

**Pull request merged**